dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -0,0 +1,59 @@
+class Api::V1::UserSchoolInfosController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_user_school_info
+  load_and_authorize_resource
+
+  # PATCH /api/v1/users_school_infos/<id>/update_last_confirmation_date
+  def update_last_confirmation_date
+    if @user_school_info.update!(last_confirmation_date: DateTime.now)
+      head :no_content
+    else
+      render json: @user_school_info.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH /api/v1/users_school_infos/<id>/update_end_date
+  def update_end_date
+    @user_school_info.update!(end_date: DateTime.now, last_confirmation_date: DateTime.now)
+    @user_school_info.user.update!(properties: {last_seen_school_info_interstitial: DateTime.now})
+
+    head :no_content
+  end
+
+  # PATCH /api/v1/user_school_infos/<id>/update_school_info_id
+  def update_school_info_id
+    ActiveRecord::Base.transaction do
+      school = @user_school_info.school_info.school
+      unless school
+        school = School.create!(new_school_params)
+        school.school_info.create!(school_info_params)
+      end
+
+      school_info = school.school_info.order(created_at: :desc).first
+
+      user = @user_school_info.user
+
+      user.user_school_infos.create!({school_info_id: school_info.id, last_confirmation_date: DateTime.now, start_date: user.created_at})
+
+      if user.update({school_info_id: school_info.id})
```

Comment:
I think you want `user.update({school_info_id: school_info.id}).valid?` here?

Reply:
Thanks!  I decided to go with the option of update!, which will throw an error if the record is not successfully saved.